### ISSUE TITLE
Switch dropdowns to use semantic <ul> and <li> markup for accessibility

### ISSUE THREAD

**Issue by tmorehouse** [bootstrap-vue/bootstrap-vue#3072]
Bootstrap V4 is moving back to recommending semantic markup for their dropdown menus.

Placing the dropdown items (and other sub-components) inside `<li>`  elements, and settign the dropdown menu container to a `<ul>`

https://github.com/twbs/bootstrap/pull/28591
https://github.com/twbs/bootstrap/issues/25728

We should update these components to use the new recommended markup.

**Comment by cybercussion** [comment]
I still see anchors getting used.  Am I missing something??

https://getbootstrap.com/docs/4.1/components/dropdowns/

**Comment by tmorehouse** [comment]
It wraps the anchors (or buttons, or text/form content) inside an `<li>`.

You should be seeing:

```html
<li><a role="menuitem" target="_self" href="#" class="dropdown-item">Action</a></li>
```

**Comment by cybercussion** [comment]
Mines resulting in:

```
<li><span class="flag-icon flag-icon-ca"> </span>  Français (Canada)</li>
```

From source:

```
<b-nav-item-dropdown id="language" extra-toggle-classes="nav-link-custom" right>
    <template slot="button-content">
        <span v-html="$i18n.messages[$i18n.locale].title"></span>
    </template>
    <template v-for="locale in activeLocales">
        <b-dropdown-item v-on:click="setLocale(locale.id, $event)" v-html="locale.title"
            v-bind:key="locale.id"></b-dropdown-item>
    </template>
</b-nav-item-dropdown>
```

**Comment by tmorehouse** [comment]
That is because you are using `v-html` directly on the `<b-dropdown-item>`, which trashes any inner structure.

try this:
```html
    <template v-for="locale in activeLocales">
        <b-dropdown-item v-on:click="setLocale(locale.id, $event)" v-bind:key="locale.id">
          <span v-html="locale.title"></span>
        </b-dropdown-item>
    </template>
```

**Comment by cybercussion** [comment]
Hi,
Ok gave that a shot and it works. The HTML I was writing was also a span so the result is:

```html
<li>
  <a role="menuitem" target="_self" href="#" class="dropdown-item">
    <span>
      <span class="flag-icon flag-icon-ca"> </span>  Français (Canada)
    </span>
  </a>
</li>
```

This worked ok in rc.18 (before).

**Comment by tmorehouse** [comment]
But now with the outer wrapper of `<LI>` wrapped around the items (for accessibility reasons) it won't work.

The reason you should be cautious when using v-html on components is that if something inside the component causes it to re-render it's contents, it will trash your v-html until the parent component updates for some reason. The other reason is that you shouldn't assume a certain structure inside 3rd party components (i.e. components that are not your own custom components)

**Comment by cybercussion** [comment]
@tmorehouse thanks for the tips.  I don't see `<ul>` or `<li>` in BS 4.1's examples currently and I think much of my SCSS was done w/o those tags so something is dorking up the display further.  Possible 4.3.1 docs are just behind.  I found a rogue `li { } ` bit of markup that was screwing up the switch to `<li>` tags so it looks correctly now.  It was causing width issues making me think something was wrong with the markup.

I played with doing this instead:

```html
<template v-for="locale in activeLocales">
        <b-dropdown-item v-on:click="setLocale(locale.id, $event)" v-bind:key="locale.id">
            <span class="flag-icon" v-bind:class="locale.flag"></span> {{ locale.name }}
        </b-dropdown-item>
    </template>
```
To get rid of the v-html.  Thanks for that tip.

So to circle back to [BS 4.3.1 dropdowns](https://getbootstrap.com/docs/4.3/components/dropdowns/) I don't see the `ul` or `li` examples like stated on this ticket.

**Comment by tmorehouse** [comment]
Refer to the links to Bootstrap Repo issue and Pull request (as mentioned above https://github.com/bootstrap-vue/bootstrap-vue/issues/3072#issue-431322462) for the forthcoming documentation updates at Bootstrap for V4.3.2:

twbs/bootstrap#28591
twbs/bootstrap#25728

**Comment by cybercussion** [comment]
Ok yeah, I got confused because I just went from BS 3 to 4 and felt like I was undoing things I just implemented.  Thanks for clarifying and all the tips.  I was coding to close to the sun.
